### PR TITLE
Bump shipwright build 0.12.0

### PR DESCRIPTION
**Pull request by jkhelil**
# Changes
- Update shipwright build with version 0.12.0
- Add missing operator service account rbac required by webhook serviceaccount 
- fixes webhook cert handling:
  * Certificate dnsName needs dnsNames injection through manifestival transformation due to presence of the namespace in the domain name (not handled natively by manifestival)
  * CRD caBundle injection through cert manager annotation needed to be done through manifestival transformation and due to namespace presence (not handled natively by manifestival)
  * Transformation applies only for platform other than Openshift as Openshift has its own service cert management
 

# Submitter Checklist

- [ ] Includes tests if functionality changed/was added
- [ ] Includes docs if changes are user-facing
- [ ] [Set a kind label on this PR](https://prow.k8s.io/command-help#kind)
- [ ] Release notes block has been filled in, or marked NONE

See [the contributor guide](https://github.com/shipwright-io/build/blob/main/CONTRIBUTING.md)
for details on coding conventions, github and prow interactions, and the code review process.

# Release Notes
- Update shipwright build with version 0.12.0
- Add missing operator service account rbac required by webhook serviceaccount 
- Fixes webhook cert handling: 
  * Certificate dnsName needs dnsNames injection through manifestival transformation
  * CRD caBundle injection through cert manager annotation needed to be done through manifestival transformation and due to namespace presence
  * Transformation applies only for platform other than Openshift as Openshift has its own service cert management

### CODE REVIEW

**Comment by jkhelil** [comment]
@shipwright-io/operator-reviewers

File: kodata/certificates.yaml [hunk 1]
```diff
@@ -15,7 +15,7 @@ metadata:
 spec:
   dnsNames:
   - shipwright-build-webhook.shipwright-build.svc
-  - shipwright-build-webhook.shipwright-build.svc.cluster.local
+  - shp-build-webhook.shipwright-build.svc
```
Comment (SaschaSchwarze0) [inline kodata/certificates.yaml:18]: Can you explain this ? The [service](https://github.com/shipwright-io/build/blob/main/deploy/402-service-webhook.yaml) is named "shp-build-webhook". Should not there only that be in? Also, are you sure that we never need cluster.local ? I never questioned it, but we basically always create certificates for our webhooks with svcname, svcname.namespace, svcname.namespace.svc, and svcname.namespace.svc.cluster.local like in https://github.com/shipwright-io/build/blob/main/hack/setup-webhook-cert.sh#L32-L35.

File: pkg/certmanager/cert_manager.go [hunk 2]
```diff
@@ -4,12 +4,20 @@ import (
...
 	crdclientv1 "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset/typed/apiextensions/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+)
 
-	"github.com/shipwright-io/operator/pkg/common"
+var (
+	certDomainsTemplate = []string{
+		"shipwright-build-webhook.%s.svc",
+		"shp-build-webhook.%s.svc",
```
Comment (SaschaSchwarze0) [inline pkg/certmanager/cert_manager.go:19]: Same here, I assume we only need shp-build-webhook.

Comment (jkhelil) [reply pkg/certmanager/cert_manager.go:19]: same here, I will keep just shp-build-webhook

Comment (jkhelil) [reply pkg/certmanager/cert_manager.go]: fixed

File: kodata/certificates.yaml (hunk 1 above)
Comment (jkhelil) [reply kodata/certificates.yaml]: fixed

**Review by SaschaSchwarze0 — APPROVED** [review]
/apprope
/lgtm

**Review by SaschaSchwarze0 — APPROVED** [review]
/approve
/lgtm

**Pull request merged**